src/share/poudriere/jail.sh

```diff
@@ -226,6 +229,15 @@ update_jail() {
 		make -C ${SRC_BASE} delete-old delete-old-libs DESTDIR=${JAILMNT} BATCH_DELETE_OLD_FILES=yes
 		markfs clean ${JAILMNT}
 		;;
+	git*)
+		install_from_git version_extra
+		RELEASE=$(update_version "${version_extra}")
+		jset ${JAILNAME} scm_branch "${SCM_BRANCH}"
+		jset ${JAILNAME} scm_url "${SCM_URL}"
+		update_version_env "${RELEASE}"
+		make -C ${SRC_BASE} delete-old delete-old-libs DESTDIR=${JAILMNT} BATCH_DELETE_OLD_FILES=yes
+		markfs clean ${JAILMNT}
+		;;
 	svn*)
```

Comment:
All of the special handling in the SVN branch will be need to be merged with the git changes as well.